File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/_3_ConformingContainer/Core/AcmeProcessingWorkflow.cs
using ConformingContainerAntipattern._3_ConformingContainer.Interfaces;
using ConformingContainerAntipattern._3_ConformingContainer.Outbound;
using ConformingContainerAntipattern._3_ConformingContainer.Services;

namespace ConformingContainerAntipattern._3_ConformingContainer.Core
{
  internal interface IProcessingWorkflow
  {
    void SetOutbound(IOutbound outbound);
    void ApplyTo(AcmeMessage message);
  }

  class AcmeProcessingWorkflow : IProcessingWorkflow
  {
    private readonly IRepository _repository;
    private readonly IAuthorization _authorizationRules;
    private IOutbound _outbound;

    public AcmeProcessingWorkflow()
    {
      _authorizationRules = ApplicationRoot.Context.Resolve<IAuthorization>();
      _repository = ApplicationRoot.Context.Resolve<IRepository>();
    }

    public void SetOutbound(IOutbound outbound)
    {
      _outbound = outbound;
    }

    public void ApplyTo(AcmeMessage message)
    {
      message.AuthorizeUsing(_authorizationRules);
      _repository.Save(message);
      _outbound.Send(message);
    }
  }
}

File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/Core/AcmeProcessingWorkflow.java
package ServiceLocatorAntipattern.Core;

import ServiceLocatorAntipattern.ApplicationRoot;
import ServiceLocatorAntipattern.Interfaces.Message;
import ServiceLocatorAntipattern.Outbound.Outbound;
import ServiceLocatorAntipattern.Services.IAuthorization;
import ServiceLocatorAntipattern.Services.IRepository;

public class AcmeProcessingWorkflow implements ProcessingWorkflow {
  private final IRepository _repository;
  private final IAuthorization _authorizationRules;
  private Outbound _outbound;

  public AcmeProcessingWorkflow() {
    _authorizationRules = ApplicationRoot.context.getComponent(IAuthorization.class);
    _repository = ApplicationRoot.context.getComponent(IRepository.class);
  }

  public void setOutbound(Outbound outbound) {
    _outbound = outbound;
  }

  public void applyTo(Message message) {
    message.authorizeUsing(_authorizationRules);
    _repository.save(message);
    _outbound.send(message);
  }
}
